### ISSUE TITLE
NOT WORKING FOR SYMFONY 4.2

### ISSUE THREAD

**Issue by xavierkaitha94** [GeniusesOfSymfony/WebSocketBundle#367]
Please update this to work with the latest version of symfony,
when I use this command "composer require gos/web-socket-bundle" it is returning an error with error code 1.

**Comment by mbabker** [comment]
It actually does work just fine with Symfony 4.2, using either an older Symfony Standard install (which is what the [demo app](https://github.com/GeniusesOfSymfony/WebsocketAppDemo) is based on for now) or Symfony Flex.

What errors are you getting?

**Comment by MarcelMC** [comment]
Warning: Illegal string offset 'callback'

**Comment by mbabker** [comment]
A stack trace would be helpful here.

**Comment by stipic** [comment]
Yes it works and it works great. Documentation needs some update but generally it's work. 

Give us some stack trace / your composer.json / dev.logs ...

**Comment by dark-matter-matters** [comment]
I can confirm this issue with Symfony 4.2. 
Here's the stack trace for composer require gos/web-socket-bundle
`Using version ^1.8 for gos/web-socket-bundle
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Restricting packages listed in "symfony/symfony" to "4.2.*"
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - gos/web-socket-bundle v1.8.9 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.8 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.7 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.6 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.5 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.4 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.3 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.2 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.11 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.10 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.1 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - gos/web-socket-bundle v1.8.0 requires symfony/framework-bundle ~2.3|~3.0 -> no matching package found.
    - Conclusion: don't install gos/web-socket-bundle v1.8.13`

**Comment by stipic** [comment]
My recommended versions for symfony 4.2

"gos/pubsub-router-bundle": "0.3.2"
"gos/web-socket-bundle": "1.9.0-rc3"

**Comment by mohamedaymenkarmous** [comment]
I found the problem after hours of searching.

You have to check the installed "gos/web-socket-bundle" version.

I though that I'm working with the master branch (2.*@dev) but since requirements are not met, I found that I'm working with 1.8 version.

So please check the documentation related to the used version.

According to 1.x version, rpc handlers are defined like this:

```
acme_rpc:
    channel: sample/{method}
    handler:
        callback: 'acme.rpc' #related to the getName() or your RPC service
    requirements:
        method:
            pattern: "[a-z_]+"
```

Note the "callback" child (not like master branch: 2.*@dev)

[Source](https://github.com/GeniusesOfSymfony/WebSocketBundle/blob/1.x/Resources/docs/RPCSetup.md)

I hope this could help
